[6.x] Improve delete states, and map them to variables

jaygeorge opened the pull request:
This PR adds delete/danger and success states and maps them to variables such as red and green.

My goal was to style these states in a way that they're obvious but not garish.

These states are also now mapped to variables and kick in automatically when aria keywords are used on buttons such as  "Delete", "Clear", "Remove", and "Reset". (As part of this I've sprinkled a few aria-roles where we have just an icon by itself)

Here are some screenshots of the delete styles:

Deleting tags is now very obvious

![2025-11-27 at 21 58 08@2x](https://github.com/user-attachments/assets/8daefd03-5c79-4159-b9a4-dfdb201ef082)

Column and row deletion is highlighted

![2025-11-27 at 21 59 18@2x](https://github.com/user-attachments/assets/1af94e68-890a-44f9-b9a9-7f6711bb6b8b)

Example of a clear selection button hover/focus state

![2025-11-27 at 22 00 34@2x](https://github.com/user-attachments/assets/a357b72c-ed34-4e94-bfa3-47e8b9ad4c7c)

Example of dark mode adjustments

![2025-11-27 at 22 02 38@2x](https://github.com/user-attachments/assets/6e4ecaec-a89a-4124-a7ad-d7110d7a3660)

jaygeorge committed:
bump up "subtle" button opacity slightly based the "x" controls on tables feeling a bit too faint

jaygeorge committed:
Start adding red delete controls

jaygeorge committed:
Add some default "red button states" for subtle buttons that have a delete control

jaygeorge committed:
Adding red delete controls for the table fieldtype

jaygeorge committed:
Finish adding red delete controls for the table fieldtype

jaygeorge committed:
Add red delete controls for combobox tags fieldtype

jaygeorge committed:
Typo

jaygeorge committed:
Replace deletes reds with a variable to tidy things

jaygeorge committed:
Dark mode adjustments

jaygeorge committed:
Use Tailwind good and proper

jaygeorge committed:
Delete buttons - Switch from transparent background values because sometimes the backgrounds stack up and you get undesirable darker colors, and fix .dark overrides for @theme inline vars, based on using a dark mode background for delete buttons

jaygeorge committed:
Shift delete states to the utility layer so they override tailwind utility classes, which we want

jaygeorge committed:
Delete buttons - date pickers and comboboxes

jaygeorge committed:
Delete buttons - add switches as an exception

jaygeorge committed:
Delete buttons - add background to dropdown

jaygeorge committed:
Delete buttons - clearable

jaygeorge committed:
Delete states - use outline state instead so we don't have to worry about overriding borders, or adding borders to pseudo elements

jaygeorge committed:
Delete states - time picker

jaygeorge committed:
Delete states - change the way we handle badges to make this clearer

jaygeorge committed:
Delete states - change the way we handle badges to make this clearer

jaygeorge committed:
Delete states - dim warning color for accessibility contrast

jaygeorge committed:
Rename color-warning to color-danger
I think the convention is warning = yellow, red = danger

jaygeorge committed:
Rename color-warning to color-danger

jaygeorge committed:
Delete states - Fix dropdown since I've changed this from "warning" to "danger"
My suspicion is that we now need !important because TW compiles classes alphabetically and "danger" will come earlier in the cascade vs "warning"

jaygeorge committed:
Delete states - take danger and success from the theme variables instead

jaygeorge committed:
Delete states - fix ui badge focus states

jaygeorge committed:
Delete states - add condition for color fieldtype

jaygeorge committed:
Fix the @theme inline color system

jaygeorge committed:
Delete states - Reorganise a little

jaygeorge committed:
Delete states - Tweak some values and add more

jaygeorge committed:
Correct gray-950 value

jaygeorge committed:
Delete states - get dark mode working

jaygeorge committed:
Delete states - simplify configuration

jaygeorge committed:
Delete states - fix missing colors

jaygeorge committed:
Delete states - fix offset

jaygeorge committed:
Delete states - Remove @theme inline rule that won't work

jaygeorge committed:
Delete states - Remove some unnecessary tw classes

jaygeorge committed:
Delete states - make subtler

jaygeorge committed:
Delete states - more work including dark mode

jaygeorge committed:
Delete states - add a load of aria labels that will trigger the delete state

jaygeorge committed:
Delete states - remove dark variants since we generate them automatically

jaygeorge committed:
Reorganise file slightly

jaygeorge committed:
Whoops, we actually need the dark variants

jaygeorge committed:
Unify table delete states so both hover and focus are the same. I can't see a reason to split them

jasonvarga reviewed (changes requested):
Using the aria label as a selector will break if you're using the CP in a language other than English. e.g. When using French you'd have html like `aria-label="Supprimer"` instead of `aria-label="Delete"`.

jaygeorge committed:
Progress making delete states explicit rathen than relying on aria labels

jaygeorge committed:
Try switching from @theme-inline to native CSS vars to get dark variants working when not descended from dark mode (the root)
For example, I was having trouble switching `/cp/assets > edit > delete button` to switch to the dark variant

jaygeorge committed:
Add additional dark selector to the theme variables. This is needed when targeting things descended from .dark panels, for example `/cp/assets > edit > delete button` , which should be a dark variant

jaygeorge committed:
Fix danger vars

jaygeorge committed:
Add explicit delete-action to other buttons

jaygeorge committed:
Remove border-radius override, decided against it

jaygeorge committed:
Fix badges

jaygeorge committed:
Add dictionary fieldtype

jaygeorge committed:
Rename delete-action to destructive

jaygeorge committed:
Merge branch 'master' into improve-delete-hover-focus

accepting master Conflicts in:
	resources/js/components/blueprints/ImportField.vue
	resources/js/components/fieldtypes/ListFieldtype.vue

jaygeorge committed:
Fix previous conflicts

jaygeorge committed:
Merge branch 'master' into improve-delete-hover-focus

Conflicts:
resources/js/components/ui/Combobox/Combobox.vue

jaygeorge committed:
Fix previous conflict

jaygeorge committed:
Merge branch 'master' into improve-delete-hover-focus

jaygeorge committed:
Remove table background color, I think it's a little much on reflection

jaygeorge committed:
Remove row background when focused or hovered

jaygeorge commented:
Closing this in favour of https://github.com/statamic/cms/pull/14220
